packages/maybe/src/combineArray.ts

```diff
@@ -0,0 +1,73 @@
+import { Just } from './Just'
+import { Maybe } from './Maybe'
+import { Nothing } from './Nothing'
+import { curry2 } from '@typed/functions'
+import { fromJust } from './fromJust'
+import { isNothing } from './isNothing'
+
+/**
+ * Applies a function with all of the values contained in an array of `Maybe`s.
+ * If *any* of the `Maybe`s are `Nothing`s then `Nothing` is returned.
+ * @name combineArray<R>(f: (...values: Array<any>) => R, maybes: ReadonlyArray<Maybe<any>>): R
+ */
+export const combineArray: CombineArray = curry2(__combineArray)
+
+function __combineArray<R>(
+  f: (...values: Array<any>) => R,
+  maybes: ReadonlyArray<Maybe<any>>
+): Maybe<R> {
+  const containsNothing = maybes.some(isNothing)
+
+  return containsNothing
+    ? Nothing
+    : Just.of<R>(f.apply(null, (maybes as ReadonlyArray<Just<any>>).map(fromJust)))
```

Comment:
Optimization? (I hate the `null`)

Reply:
Not in particular. I suppose it could either be `void 0` or it could actually pass along whatever `this` context is available to it.